Modules/_cursesmodule.c

```diff
@@ -3024,16 +3147,24 @@ _curses_has_key_impl(PyObject *module, int key)
 }
 #endif
 
+#if _NCURSES_EXTENDED_COLOR_FUNCS
+#define _CURSES_INIT_COLOR_FUNC         init_extended_color
+#else
+#define _CURSES_INIT_COLOR_FUNC         init_color
+#endif  /* _NCURSES_EXTENDED_COLOR_FUNCS */
+
+#define _CURSES_INIT_COLOR_FUNC_NAME    _CURSES_FUNC_NAME_STR(_CURSES_INIT_COLOR_FUNC)
+
```

Comment:
It adds 11 lines of code (8 lines to define `_CURSES_INIT_COLOR_FUNC` and `_CURSES_INIT_COLOR_FUNC_NAME` and other 3 lines to undefine them). But using `#if/#else/#endif` at the call place would add only 4 lines of code. Would not it be simpler?

The same question is about other defines used only once.

Reply:
Other option is to define all macros in a single `#if/#else/#endif` construction:
```c
#if _NCURSES_EXTENDED_COLOR_FUNCS
#  define _NCURSES_COLOR_VAL_TYPE   int
#  define _COLOR_CONTENT_FUNC       extended_color_content
#  define _CURSES_INIT_COLOR_FUNC   init_extended_color
#  define _CURSES_INIT_PAIR_FUNC    init_extended_pair
#  define _CURSES_PAIR_NUMBER_FUNC  extended_pair_content
#else
...
```

Reply:
I went with the latter. It seems more clear than leaving it scattered throughout the code.